docs/en/ingest-management/fleet/air-gapped.asciidoc

```diff
@@ -32,6 +32,48 @@ Set the following property in {kib} to enable air-gapped mode in {fleet}. This a
 xpack.fleet.isAirGapped: true
 ----
 
+[discrete]
+[[air-gapped-pgp-fleet]]
+== Configure {agents} to download a PGP/GPG key from {fleet-server}
+
+Starting from version 8.9.0, when {agent} tries to perform an upgrade, it first verifies the binary signature with the key bundled in the agent. This process has a backup mechanism that will use the key coming from `https://artifacts.elastic.co/GPG-KEY-elastic-agent` instead of the one it already has.
+
+In an air-gapped environment, the agent won't be able to download the remote key and therefore cannot be upgraded. For versions 8.9.0 to 8.10.3, you can resolve this problem following the steps described in the associated link:https://www.elastic.co/guide/en/fleet/8.9/release-notes-8.9.0.html#known-issues-8.9.0[known issue] documentation.
+
+Starting in version 8.10.4, you can resolve this problem by configuring {agents} to download the PGP/GPG key from a custom URL specified in {fleet-server}:
```

Comment:
I think my reading comprehension on my 1st pass wasn't great.

Just to be clear fleet-server serves PGP keys at `GET /api/agents/upgrades/{major}.{minor}.{patch}/pgp-public-key`
It will serve the key in `FLEETSERVER_BINARY_DIR/elastic-agent-upgrade-keys/default.pgp` by default, if there is no `default.pgp` key in the dir it will look upstream to retrieve a key; `server.pgp.upstream_url`  can be used to configure to a custom URL for fleet-server to use.

IIRC agents will try to get the upstream key, then the fleet-server endpoint (@michalpristas please confirm), they do not use the url we add to fleet-server config

Reply:
Thanks @michel-laterman! That's really clear. Here's my attempt to fix things up:

---

![Screenshot 2024-07-30 at 12 37 29 PM](https://github.com/user-attachments/assets/7d541046-db25-495c-9326-95dd30e9e3ee)